File: app/src/main/java/com/afterroot/photos/CustomFragmentActivity.java
package com.afterroot.photos;

import android.app.Fragment;
import android.app.FragmentManager;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;

import com.afterroot.photos.fragment.SettingsFragment;

public class CustomFragmentActivity extends AppCompatActivity {

    public static final int SETTINGS_FRAGMENT = 2;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_custom_fragment);

        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setTitle(getFragmentTitle());

        initFragments();
    }

    public void initFragments(){
        switch (getFragmentId()){
            case SETTINGS_FRAGMENT:
                SettingsFragment settingsFragment = new SettingsFragment();
                setFragment(settingsFragment);
                break;
        }
    }

    public String getFragmentTitle(){
        return getIntent().getStringExtra(getString(R.string.key_fragment_title));
    }

    private int getFragmentId(){
        int extra = getIntent().getIntExtra(getString(R.string.key_fragment_id), 0);
        Log.d(getString(R.string.key_fragment_id), "id: " + extra);
        return extra;
    }

    FragmentManager mFragmentManager;
    private void setFragment(Fragment fragment){
        if (mFragmentManager == null){
            mFragmentManager = getFragmentManager();
            mFragmentManager.beginTransaction().replace(R.id.content_custom_fragment, fragment).commit();
        } else {
            mFragmentManager.beginTransaction().replace(R.id.content_custom_fragment, fragment).commit();
        }
    }
}

File: app/src/main/java/com/afterroot/tagit/CustomFragmentActivity.kt
package com.afterroot.tagit

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import com.afterroot.tagit.fragment.MainFragment
import com.afterroot.tagit.fragment.SettingsFragment

class CustomFragmentActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_custom_fragment)

        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        supportActionBar!!.title = fragmentTitle

        setFragments()
    }

    private fun setFragments() {
        when (fragmentId) {
            SETTINGS_FRAGMENT -> setFragment(SettingsFragment())
            MAIN_FRAGMENT -> setFragment(MainFragment())
        }
    }

    private val fragmentTitle: String?
        get() = intent.getStringExtra(getString(R.string.key_fragment_title))

    private val fragmentId: Int
        get() {
            val extra = intent.getIntExtra(getString(R.string.key_fragment_id), 0)
            Log.d(getString(R.string.key_fragment_id), "id: $extra")
            return extra
        }

    private var mFragmentManager: androidx.fragment.app.FragmentManager? = null
    private fun setFragment(fragment: androidx.fragment.app.Fragment) {
        if (mFragmentManager == null) {
            mFragmentManager = supportFragmentManager
        }
        mFragmentManager!!.beginTransaction().replace(R.id.content_custom_fragment, fragment).commit()
    }

    companion object {
        const val SETTINGS_FRAGMENT = 2
        const val MAIN_FRAGMENT = 1
    }
}
